pkg/registry/helm.go

```diff
@@ -0,0 +1,297 @@
+}
+
+func helmRepositoryName(repoURL, tempRepo, chart string) string {
+	if strings.HasPrefix(repoURL, "http") {
+		return fmt.Sprintf("%s/%s", tempRepo, chart)
+	}
+
+	return repoURL
+}
+
+func helmAddRepoCommand(repo, tempRepo string) string {
+	if !strings.HasPrefix(repo, "http") {
+		return ""
+	}
+
+	return fmt.Sprintf("helm repo add %s %s", tempRepo, repo)
+}
+
+func helmPullCommand(repository, chart, version string, destDir string) string {
+	repository = helmRepositoryName(repository, fmt.Sprintf("repo-%s", chart), chart)
+
+	pullCommand := fmt.Sprintf("helm pull %s", repository)
+	if version != "" {
+		pullCommand = fmt.Sprintf("helm pull %s --version %s", repository, version)
+	}
+	if destDir != "" {
+		pullCommand = fmt.Sprintf("%s -d %s", pullCommand, destDir)
+	}
+
+	return pullCommand
+}
+
+func helmTemplateCommand(crd *HelmCRD, repository string, valuesFilePath string, chartName string) string {
+	var cmdParts []string
+	cmdParts = append(cmdParts, fmt.Sprintf("helm template --skip-crds %s %s", chartName, repository))
+
+	if crd.Spec.Version != "" {
+		cmdParts = append(cmdParts, fmt.Sprintf("--version %s", crd.Spec.Version))
+	}
+
+	if len(crd.Spec.Set) > 0 {
+		args := parseSetArgs("", crd.Spec.Set)
+		cmdParts = append(cmdParts, fmt.Sprintf("--set %s", strings.Join(args, ",")))
+	}
+
+	if crd.Spec.ValuesContent != "" {
+		cmdParts = append(cmdParts, fmt.Sprintf("-f %s", valuesFilePath))
+	}
+
+	return strings.Join(cmdParts, " ")
+}
+
+func updateHelmManifest(manifestPath string, chartTarsPaths []string) ([]map[string]any, error) {
+	manifestFile, err := os.ReadFile(manifestPath)
+	if err != nil {
+		return nil, fmt.Errorf("reading helm manifest '%s': %w", manifestPath, err)
+	}
+
+	var manifests []map[string]any
+	decoder := yaml.NewDecoder(bytes.NewReader(manifestFile))
+	for {
+		var manifest map[string]any
+
+		if err = decoder.Decode(&manifest); err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			return nil, fmt.Errorf("unmarshaling manifest '%s': %w", manifestPath, err)
+		}
+
+		kind, ok := manifest["kind"]
+		if !ok || kind != HelmChartKind {
+			manifests = append(manifests, manifest)
+			continue
+		}
+
+		if spec, ok := manifest["spec"].(map[string]any); ok {
+			if _, ok := spec["chartContent"].(string); ok {
+				manifests = append(manifests, manifest)
+				continue
+			}
+			oldChart := spec["chart"]
+			delete(spec, "repo")
+			delete(spec, "chart")
+			for _, chartTar := range chartTarsPaths {
+				if strings.Contains(chartTar, oldChart.(string)) {
+					tarData, err := os.ReadFile(chartTar)
+					if err != nil {
+						return nil, fmt.Errorf("reading chart tar '%s': %w", chartTar, err)
+					}
+					base64Str := base64.StdEncoding.EncodeToString(tarData)
+					spec["chartContent"] = base64Str
+				}
+			}
+		}
+
+		manifests = append(manifests, manifest)
+	}
+
+	return manifests, nil
+}
+
+func UpdateAllManifests(localHelmSrcDir string, chartTarsPath []string) ([][]map[string]any, error) {
```

Comment:
What do we refer to with "All" here? Can we rename this to `UpdateHelmManifests`?